src/main/webapp/app/guided-tour/guided-tour.component.ts

```diff
@@ -173,68 +178,191 @@ export class GuidedTourComponent implements AfterViewInit, OnDestroy {
      * @param element that should be checked
+     * @param direction it should be checked if the tour step is horizontally or vertically in the viewport
      * @return true if element is in viewport, otherwise false
      */
-    public elementInViewport(element: HTMLElement | null): boolean {
+    private elementInViewport(element: HTMLElement | null, direction: Direction): boolean {
         if (!element) {
             return false;
         }
-        let top = element.offsetTop;
-        const height = element.offsetHeight;
 
-        while (element.offsetParent) {
-            element = element.offsetParent as HTMLElement;
-            top += element.offsetTop;
-        }
+        let elementInViewPort = true;
 
-        const scrollAdjustment = this.currentTourStep && this.currentTourStep.scrollAdjustment ? this.currentTourStep.scrollAdjustment : 0;
-        const stepScreenAdjustment = this.getStepScreenAdjustment();
+        switch (direction) {
+            case Direction.HORIZONTAL: {
+                const width = element.offsetWidth;
+                const left = calculateLeftOffset(element);
+                const tourStepWidth = this.tourStep.nativeElement.offsetWidth;
+                elementInViewPort = isElementInViewPortHorizontally(this.currentTourStep.orientation, left, width, tourStepWidth);
+                break;
+            }
+            case Direction.VERTICAL: {
+                const scrollAdjustment = this.currentTourStep && this.currentTourStep.scrollAdjustment ? this.currentTourStep.scrollAdjustment : 0;
+                const stepScreenAdjustment = this.getStepScreenAdjustment();
+                const top = calculateTopOffset(element);
+                const height = element.offsetHeight;
+
+                if (this.isBottom()) {
+                    elementInViewPort =
+                        top >= window.pageYOffset + this.topOfPageAdjustment + scrollAdjustment + stepScreenAdjustment && top + height <= window.innerHeight + window.pageYOffset;
+                } else {
+                    elementInViewPort =
+                        top >= window.pageYOffset + this.topOfPageAdjustment - stepScreenAdjustment && top + height + scrollAdjustment <= window.innerHeight + window.pageYOffset;
+                }
+                break;
+            }
+        }
+        return elementInViewPort;
+    }
 
-        if (this.isBottom()) {
-            return top >= window.pageYOffset + this.topOfPageAdjustment + scrollAdjustment + stepScreenAdjustment && top + height <= window.innerHeight;
-        } else {
-            return top >= window.pageYOffset + this.topOfPageAdjustment - stepScreenAdjustment && top + height + scrollAdjustment <= window.innerHeight;
+    /**
+     * Flips the orientation of the current tour step horizontally
+     */
+    private flipOrientation(): void {
+        if (this.isLeft()) {
+            switch (this.orientation) {
```

Comment:
Hm here you are accessing `orientation` directly on the class and below you do it on `currentTourStep`. If I understand correctly it should be only accessed with the first approach.

Reply:
You are right, I missed that part. I changed the code to `switch (this.currentTourStep.orientation)` now